src/Components/Users/UserProfile.tsx

```diff
@@ -0,0 +1,397 @@
+        let form = { ...states.form };
+        form[e.target.name] = e.target.value;
+        dispatch({ type: "set_form", form });
+    };
+
+
+    const validateForm = () => {
+        let errors = { ...initError };
+        let invalidForm = false;
+        Object.keys(states.form).forEach((field, i) => {
+            switch (field) {
+                case "first_name":
+                case "last_name":
+                case "gender":
+                    if (!states.form[field]) {
+                        errors[field] = "Field is required";
+                        invalidForm = true;
+                    }
+                    return;
+                case "age":
+                    if (!states.form[field]) {
+                        errors[field] = "This field is required";
+                        invalidForm = true;
+                    }
+                    return;
+                case "phone_number":
+                    const phoneNumber = parsePhoneNumberFromString(states.form[field]);
+                    if (!states.form[field] || !phoneNumber?.isPossible()) {
+                        errors[field] = "Please enter valid phone number";
+                        invalidForm = true;
+                    }
+                    return;
+                case "email":
+                    if (states.form[field] && !validateEmailAddress(states.form[field])) {
+                        errors[field] = "Enter a valid email address";
+                        invalidForm = true;
+                    }
+                    return;
+                default:
+                    return;
+            }
+        });
+        dispatch({ type: "set_error", errors });
+        return !invalidForm;
+    };
+
+
+    const handleValueChange = (phoneNo: any, name: string) => {
+        if (phoneNo && parsePhoneNumberFromString(phoneNo)?.isPossible()) {
+            const form = { ...states.form };
+            form[name] = phoneNo;
+            dispatch({ type: "set_form", form });
+        }
+    };
+
+
+    const handleSubmit = async (e: any) => {
+        e.preventDefault();
+        const validForm = validateForm();
+        if (validForm) {
+            setIsLoading(true);
+            const data = {
+                username: username,
+                first_name: states.form.first_name,
+                last_name: states.form.last_name,
+                email: states.form.email,
+                phone_number: parsePhoneNumberFromString(states.form.phone_number)?.format('E.164'),
+                gender: Number(states.form.gender),
+                age: states.form.age,
+
+
+            };
+            const res = await dispatchAction(
+                updateUserDetails(username, data)
+            );
+            setIsLoading(false);
+            if (res && res.data) {
+                Notification.Success({
+                    msg: "Details updated successfully"
+                });
+                window.location.reload();
```

Comment:
@aashnanaushad It's not adviced to use `window.location.reload();` as you have all the data from the forms, you could just update the state, its what react is for :smile: